Don't translate point for independent stacking contexts

fiji-flo opened the pull request:
I took the test for hit_test from:
https://github.com/servo/servo/pull/15047/commits/602b45a3724747e1be767605c523c510846cfbf6
Thank you @emilio !

<!-- Please describe your changes on the following line: -->
Don't keep the translation for a given point of interest when traversing independent stacking contexts.
This was an error introduced in #14291
---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix  #15015 (gith1ub issue number if applicable).

<!-- Either: -->
- [X ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15049)
<!-- Reviewable:end -->

highfive commented:
Heads up! This PR modifies the following files:
 * @emilio: components/layout/query.rs

fiji-flo committed:
Don't translate point for independent stacking contexts

I took the test for hit_test from:
https://github.com/servo/servo/pull/15047/commits/602b45a3724747e1be767605c523c510846cfbf6
Thank you emilio

fiji-flo commented:
Closing this. @emilio merged it into #15047. Thanks!
